Backport d4853d030639cf3542ae39129c18b654d8d4f020 from upstream klibc

Newer kernel headers move the definition of MAP_PRIVATE.  This breaks
the klibc build.
--- a/usr/include/sys/mman.h
+++ b/usr/include/sys/mman.h
@@ -7,7 +7,7 @@
 
 #include <klibc/extern.h>
 #include <sys/types.h>
-#include <asm/mman.h>
+#include <linux/mman.h>
 
 #define MAP_FAILED ((void *)-1)
 
